.travis.yml

```diff
@@ -31,6 +31,7 @@ before_script:
 
 script:
   - coala --non-interactive -V
+  - pip install --upgrade pytest~=4.6
```

Comment:
This should not be here, you should make changes to the requirements.txt file instead

Reply:
I also wanted to change there but the problem is that coala has requirements of pytest==3.6.4 and python requires pytest==4.6.2 for running tests. So, if I update the requirements.txt file, the command ```coala --non-interactive -V``` fails. One thing that can be done is that we can change the requirements.txt file of coala too to pytest==4.6, but I don't know what impact it would have on coala.

Reply:
Pytest version for this can be changed. coala right now has a strict pytest dependency but here we can adjust.

Reply:
Thats what I am trying to ask, that if we change the pytest version in corobo then the coala check that we do in travis CI builds would fail(I have tried this out). So updating pytest before running the tests only seemed me to be a solution that would compromise this requirement conflict.

Reply:
coala doesn't need pytest to run. It is used to run tests only. pytest is a part of coala's test-requirements and not its requirements. So the dependency surely doesn't come from coala. The failure you see is because of something else.

Reply:
I also thought that but the error clearly says that there coala requirements are not satisfied and the traceback also passes through the coala_main.py file. Logically, it doesn't seem correct that test-requirements should be satisfied for coala to run. But it seems like this. I will investigate this. By then, Can you pls have a look at these logs and suggest your views:
```
$ coala --non-interactive -V
[WARNING][06:28:39] bears setting has already been defined in section all.python. The previous setting will be overridden.
[WARNING][06:28:39] bears setting has already been defined in section all.python. The previous setting will be overridden.
[ERROR][06:28:39] There is a conflict in the version of a dependency you have installed and the requirements of coala. This may be resolved by creating a separate virtual environment for coala or running `pip3 install "importlib-metadata>=1.7.0; python_version < "3.8""`. Be aware that the latter solution might break other python packages that depend on the currently installed version.
[INFO][06:28:39] Exception was:
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/coalib/coala_main.py", line 147, in run_coala
    args=args)
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/coalib/settings/ConfigurationGathering.py", line 504, in gather_configuration
    targets=targets,
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/coalib/settings/SectionFilling.py", line 97, in fill_settings
    bear_dirs = section.bear_dirs()
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/coalib/settings/Section.py", line 147, in bear_dirs
    for bear_dir in collect_registered_bears_dirs('coalabears')]
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/coalib/collecting/Collectors.py", line 456, in collect_registered_bears_dirs
    registered_package = ep.load()
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2460, in load
    self.require(*args, **kwargs)
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2483, in require
    items = working_set.resolve(reqs, env, installer, extras=self.extras)
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (importlib-metadata 0.18 (/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages), Requirement.parse('importlib-metadata>=1.7.0; python_version < "3.8"'), {'stevedore'})
The command "coala --non-interactive -V" exited with 13.
```